tests/src/python/test_qgsdatumtransforms.py

```diff
@@ -0,0 +1,148 @@
+__copyright__ = 'Copyright 2019, The QGIS Project'
+
+from qgis.core import (
+    QgsProjUtils,
+    QgsCoordinateReferenceSystem,
+    QgsDatumTransform
+)
+from qgis.testing import (start_app,
+                          unittest,
+                          )
+from utilities import unitTestDataPath
+
+start_app()
+TEST_DATA_DIR = unitTestDataPath()
+
+
+class TestPyQgsDatumTransform(unittest.TestCase):
+
+    @unittest.skipIf(QgsProjUtils.projVersionMajor() < 6, 'Not a proj6 build')
+    def testOperations(self):
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem(),
+                                           QgsCoordinateReferenceSystem())
+        self.assertEqual(ops, [])
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem('EPSG:3111'),
+                                           QgsCoordinateReferenceSystem())
+        self.assertEqual(ops, [])
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem(),
+                                           QgsCoordinateReferenceSystem('EPSG:3111'))
+        self.assertEqual(ops, [])
+
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem('EPSG:3111'),
+                                           QgsCoordinateReferenceSystem('EPSG:3111'))
+        self.assertEqual(len(ops), 1)
+        self.assertEqual(ops[0].name, 'Inverse of Vicgrid + Vicgrid')
+        self.assertEqual(ops[0].proj, '+proj=noop')
+        self.assertEqual(ops[0].accuracy, 0.0)
+        self.assertTrue(ops[0].isAvailable)
+
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem('EPSG:3111'),
+                                           QgsCoordinateReferenceSystem('EPSG:4283'))
+        self.assertEqual(len(ops), 1)
+        self.assertEqual(ops[0].name, 'Inverse of Vicgrid')
+        self.assertEqual(ops[0].proj, '+proj=pipeline +step +inv +proj=lcc +lat_0=-37 +lon_0=145 +lat_1=-36 +lat_2=-38 +x_0=2500000 +y_0=2500000 +ellps=GRS80 +step +proj=unitconvert +xy_in=rad +xy_out=deg +step +proj=axisswap +order=2,1')
+        self.assertEqual(ops[0].accuracy, -1.0)
+        self.assertTrue(ops[0].isAvailable)
+
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem('EPSG:3111'),
+                                           QgsCoordinateReferenceSystem('EPSG:28355'))
+        self.assertEqual(len(ops), 1)
+        self.assertEqual(ops[0].name, 'Inverse of Vicgrid + Map Grid of Australia zone 55')
+        self.assertEqual(ops[0].proj, '+proj=pipeline +step +inv +proj=lcc +lat_0=-37 +lon_0=145 +lat_1=-36 +lat_2=-38 +x_0=2500000 +y_0=2500000 +ellps=GRS80 +step +proj=utm +zone=55 +south +ellps=GRS80')
+        self.assertEqual(ops[0].accuracy, 0.0)
+        self.assertTrue(ops[0].isAvailable)
+
+        # uses a grid file
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem('EPSG:4283'),
+                                           QgsCoordinateReferenceSystem('EPSG:7844'))
+        self.assertEqual(len(ops), 5)
```

Comment:
relying exactly on 5 is fragile. EPSG might add new operations

Reply:
and relying on the below particular order of ops[0], ops[1], etc. is fragile too.